File: examples/serialization/MioAlma.java
import java.io.Serializable;
import java.util.Date;

/**
 * @author amit
 *
 */
public class MioAlma implements Serializable {
	/**
	 * Generated serial UID for all versions of MioAlma objects
	 */
	private static final long serialVersionUID = -7208668478357756970L;

	private String name;
	private long uniqueId;
	private Date birthDate;
	private transient String interests;

	/**
	 * @param name
	 * @param uniqueId
	 */
	public MioAlma(String name, long uniqueId) {
		this.name = name;
		this.uniqueId = uniqueId;
		this.birthDate = new Date(); // today
		this.interests = "daydreaming";
	}

	/**
	 * @return the name
	 */
	public String getName() {
		return name;
	}

	/**
	 * @param name
	 *            the name to set
	 */
	public void setName(String name) {
		this.name = name;
	}

	/**
	 * @return the uniqueId
	 */
	public long getUniqueId() {
		return uniqueId;
	}

	/**
	 * @return the birthDate
	 */
	public Date getBirthDate() {
		return birthDate;
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		return "MioAlma [name=" + name + ", uniqueId=" + uniqueId + ", birthDate=" + birthDate + ", interests="
				+ interests + "]";
	}

}

File: examples/other-languages/go/serialization/MioAlma.go
//Author: Amit Jain & Paul Ellis

package main

import (
	"time"
	"fmt"
)

//Unlike in Java there is no keyword to privatize variables
//From serialization. The only way is to make them "private"
//Which means that you leave the first letter of the name
//lower case. Upper case variable names are "public"
type MioAlma struct{
	Name string
	UniqueID int64
	Birthdate time.Time
	interests string
}

func NewMioAlma(name string, uid int64) *MioAlma{
	m := MioAlma{Name:name, UniqueID:uid, Birthdate:time.Now(), interests:"daydreaming"}
	return &m
}

func (m *MioAlma)GetName() string{
	return m.Name
}

func (m *MioAlma)SetName(newName string){
	m.Name = newName
}

func (m *MioAlma)GetUniqueId() int64{
	return m.UniqueID
}

func (m *MioAlma)GetBirthdate() time.Time{
	return m.Birthdate
}

func (m *MioAlma)ToString() string{
	msg := fmt.Sprintf("MioAlma [name=%s, uniqueId=%d, birthdate=%s, interests=%s]",
			m.Name,
			m.UniqueID,
			m.Birthdate.Format(time.UnixDate),
			m.interests)
	return msg
}